### ISSUE TITLE
Incorrectly reporting SA1121 in Expression Bodied Function

### ISSUE THREAD

**Issue by a-jackson** [DotNetAnalyzers/StyleCopAnalyzers#2489]
I had an Expression Bodied Function creating an instance of of a type using an Object Initializer where one of the property names was `Object`. This incorrectly reporting SA1121 Use built-in type alias however the usage of `Object` was a property name rather than a type.

Changing the function to a normal method fixed the issue. See below

```c#
private ObjectVersionPair<DataAsset, DataAssetVersion> GetAssetVersionPair(AssetTemplates x) =>
    new ObjectVersionPair<DataAsset, DataAssetVersion>
    {
        Object = GetObject(x),
        Version = GetVersion(x),
    };
```
Reporting SA1121 Use built-in type alias

```c#
private ObjectVersionPair<DataAsset, DataAssetVersion> GetAssetVersionPair(AssetTemplates x)
{
    return new ObjectVersionPair<DataAsset, DataAssetVersion>
    {
        Object = GetObject(x),
        Version = GetVersion(x),
    };
}
```

Works as expected

Using StyleCop.Analyzers v1.0.2

**Comment by vweijsters** [comment]
Hi @a-jackson, I cannot reproduce the error that you describe. Could you please provide a more complete sample, including details on the used classes?

**Comment by a-jackson** [comment]
Since reporting this I've upgraded to VS 2017 and now can't reproduce it myself. It was originally in a dotnet core class library project with VS 2015, so an .xproj project.

I can attempt to make a minimal project with 2015 to try and reproduce this if you like or just close it as it no longer occurs with latest versions?

**Comment by vweijsters** [comment]
Feel free to close it if the problem no longer exists for you. 
If you create a minimal project with the problem in it, I will of course have a look 😉

**Comment by a-jackson** [comment]
I've managed to reproduce it in Visual Studio 2015. Code is here https://github.com/a-jackson/StyleCop.Analyzers.Issue2489

I migrated the project to Visual Studio 2017 and it works as expected so it seems to be limited to 2015. Seems to only occur with the `using System;` which VS recognises as not needed but that makes sense I guess as `Object` would only work as a type if there was `using System;`.

Feel free to close this though if you aren't support 2015 dotnet core projects anymore.

**Comment by vweijsters** [comment]
Thanks for the code. It reproduces here with both 1.0.2 and 1.1.0-beta004.

I cannot get it to reproduce in the development environment however 😞 

@sharwell I had a quick look at the code and this [line](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/StyleCop.Analyzers/StyleCop.Analyzers/ReadabilityRules/SA1121UseBuiltInTypeAlias.cs#L237) in SA1121 is most likely where it goes wrong. Do you have any idea how this can be reproduced in the development environment?

**Comment by pdelvo** [comment]
Do you know the version of the compiler that is used? This looks like a compiler bug that was fixed with https://github.com/dotnet/roslyn/pull/8167.

**Comment by a-jackson** [comment]
It'll be whatever comes with dotnet core 1.0.0-preview2-003131. I can't figure out how to find the roslyn version it is using. the csc.dll in there is 1.3.0.

**Comment by sharwell** [comment]
@a-jackson The fix for this first shipped with 2.0.0. This suggests it should reproduce in the original test project, but not the CSharp7 test project.

**Comment by vweijsters** [comment]
Closing this as an external bug
